docs/adrs/v3-setup-dotnet.md

```diff
@@ -0,0 +1,48 @@
+
+# Breaking changes
+- The presence of pre-installed .NET versions that are higher than version that the users specify in the setup task can be breaking for some customers, who expect only one installed .NET version on runner after using setup task. If user doesn't have .NET version specified in project file, `dotnet` will use the latest installed version instead of provided in setup task.  
+> **Note:** It is the biggest deal in this ADR.
+Previously, when user specifies .NET version, this exact version was used by `dotnet` command by default (because it was installed in the separate folder and there are no any other .NET versions in that folder)  
+In proposal, the specified version will be installed on machine but the latest version will be used by `dotnet` command by default (because specified version will be installed alongside with pre-installed .NET versions).  
+Based on [official .NET documentation](https://docs.microsoft.com/en-us/dotnet/core/versions/selection), it is expected behavior and how it works on user's local machine with multiple installed .NET versions but some customers could be broken because they already rely on current behavior.
+
+- All possible inputs will continue to work as previously except `5.x`. It is totally okay to drop it because we shouldn't allow user to rely on such input. New minor release of .NET will break their builds
+
+To avoid breaking customers, we will need to release new major task version (V3).
+
+# v3-preview
+There will be a v3-preview branch that will be created for development and testing. Any changes will first be merged into v3-preview branch. After a period of testing & verification, the v3-preview branch will be merged into the main branch and a v3 tag will be created. Any GitHub public documentation and starter workflows that mention setup-dotnet will then be updated to use v3 instead of v2:
+- [README.md](https://github.com/actions/setup-dotnet/blob/main/README.md)
+- [action.yml](https://github.com/actions/setup-dotnet/blob/main/action.yml)
+- [GitHub docs](https://docs.github.com/en/actions/guides/building-and-testing-net#using-a-specific-net-version)
+- Starter-workflow yamls: [#1](https://github.com/actions/starter-workflows/blob/main/ci/dotnet.yml#L17), [#2](https://github.com/actions/starter-workflows/blob/main/ci/dotnet-desktop.yml#L72)
+
+# Consequences
+- Customers will be able to use pre-installed .NET versions with setup-dotnet action on Windows and Ubuntu
+- Maintenance of the action will be easier due the simplier logic of handling inputs with wildcards
```

Comment:
Maintenance of the action will be easier due to the simpler logic of handling inputs with wildcards